runtime/src/mm/cpp/ExtraObjectDataTest.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+#include "ExtraObjectData.hpp"
+
+#include <atomic>
+#include <thread>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "TestSupport.hpp"
+
+using namespace kotlin;
+
+TEST(ExtraObjectDataTest, Install) {
+    TypeInfo typeInfo;
+    typeInfo.typeInfo_ = &typeInfo;
+    ObjHeader object;
+    object.typeInfoOrMeta_ = &typeInfo;
+
+    ASSERT_FALSE(object.has_meta_object());
+
+    auto& extraData = mm::ExtraObjectData::Install(&object);
+
+    EXPECT_TRUE(object.has_meta_object());
```

Comment:
Just curious: why `ASSERT_FALSE` but `EXPECT_TRUE`?

Reply:
In `googletest` there are 2 kinds of test assertions: `EXPECT_*` and `ASSERT_*`. The former record the error and move on, the latter record the error and stop the test (technically, just execute the `return` statement). So, I typically use `ASSERT_*` as a precondition: when there's no point to run the test if it does not hold.